### DISCUSSION TITLE
How to write device service by python ?

### DISCUSSION THREAD

**Discussion by omidfathi** [edgexfoundry/community#134] [Q&A]
Hi,
I write a program in python for OPCUA Client and I want to connect it to Edgex as device service.
Can you tell me is it possible? and if it is possible, how should it works and where should I start ?

**Comment by cloudxxx8** [comment]
It definitely doable, and I heard someone did this before.
There is an old document for reference: https://docs.google.com/document/d/1aMIQ0kb46VE5eeCpDlaTg8PP29-DBSBTlgeWrv6LuYk/edit
The better way is to ready the current SDK code to just understand the flow and logic.
